### ISSUE TITLE
Asynchronous Federated learning, predicting with traced model

### ISSUE THREAD

**Issue by abdulbasitds** [OpenMined/PySyft#2925]
Hello, after having the aggregated model (traced_model = utils.federated_avg(models)).how can we predict classes from it, I am doing traced_model(X_test) but it gives same dimensionalty as of input (N*2) where I expect it to be array of target class (N*1)

Here is my code 

```
for curr_round in range(1, epochs+1):
            #logger.info("Training round %s/%s", curr_round, args.training_rounds)
            results = [
                    fit_model_on_worker(
                        worker=worker,
                        traced_model=traced_model,
                        batch_size=32,
                        curr_round=curr_round,
                        max_nr_batches=10,
                        lr=learning_rate,
                    )
                    for worker in recruited_instances
                ]
            
            models = {}
            loss_values = {}
            test_models = curr_round % 10 == 1 or curr_round == epochs
            if test_models:
                #logger.info("Evaluating models")
                #print(evaluating models)
                np.set_printoptions(formatter={"float": "{: .0f}".format})
                for worker_id, worker_model, _ in results:
                    evaluate_model_on_worker(
                        model_identifier="Model update " + str(worker_id),
                        worker=testing[0],
                        dataset_key="classification_data_test",
                        model=worker_model,
                        nr_bins=10,
                        batch_size=128,
                        #device=device,
                        print_target_hist=False,
                    )
            
            for worker_id, worker_model, worker_loss in results:
                if worker_model is not None:
                    models[worker_id] = worker_model
                    loss_values[worker_id] = worker_loss

            traced_model = utils.federated_avg(models)
```
So, After having `traced_model`, I want to predict with on a set of example like

`y_pred = traced_model(X_test)`
But, it doesnt gives correct result of N*1 dimension ([0,1,1,0,1,1]) but return inputs instead.

I tried `traced,model.predict(X)` but its says scriptable object has no method predict

**Comment by abdulbasitds** [comment]
So, Since it had softmax at the output, i need to use argmax()
